lib/remote/configpackageutility.cpp

```diff
@@ -196,9 +197,15 @@ void ConfigPackageUtility::TryActivateStageCallback(const ProcessResult& pr, con
 			<< "Config validation failed for package '"
 			<< packageName << "' and stage '" << stageName << "'.";
 	}
+
+	// Don't let the deferred callback go out of scope when the config validation was successful
+	// and the Icinga2 worker process is going to be reloaded
+	if (reload && pr.ExitStatus == 0) {
+		ConfigStagesHandler::m_KeepRunning = resetPackageUpdates;
```

Comment:
Having this member just to prevent the destruction of the `Defer` feels a bit hacky. What about adding `Defer::Cancel()` which resets the captured function so that if it was called,  `~Defer()` just does nothing?